java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEngineRunner.java

```diff
@@ -103,18 +117,26 @@ public void start() throws InterruptedException {
             return;
         }
 
+        executor.execute(engine);
+        // wait for start of the connector task
+        boolean startOk = startSignal.await(5, TimeUnit.SECONDS);
```

Comment:
Any other way to wait for start of connector task? May fix later, but leave some issue to track.

Reply:
Sleep `5` seems brittle.

Reply:
Or is `5` just the timeout?